### PR TITLE
ddr support for Valhalla NullRestricted attribute

### PR DESCRIPTION
**Pull request by theresa-m**
- include internal modifiers for fields in rom class dump
- fix incorrectly ordered arguments of dumpModifers method
- print NullRestricted field modifier in ddr rom class dump

Related: https://github.com/eclipse-openj9/openj9/issues/17340

### CODE REVIEW

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: debugtools/DDR_VM/data/superset-constants.dat [hunk 1]
```diff
@@ -2116,6 +2116,7 @@ C|J9FieldFlagHasGenericSignature
 C|J9FieldFlagHasTenantAnnotation
 C|J9FieldFlagHasTypeAnnotations
 C|J9FieldFlagIsContended
+C|J9FieldFlagIsNullRestricted
```
[the remark this answers is not in the document]
Comment (theresa-m) [reply debugtools/DDR_VM/data/superset-constants.dat:2119]: Thanks, updated.

[the remark this answers is not in the document]
Comment (keithc-ca) [reply debugtools/DDR_VM/data/superset-constants.dat:2119]: Can you push so the rest of us can see your update? ;-)

[the remark this answers is not in the document]
Comment (theresa-m) [reply debugtools/DDR_VM/data/superset-constants.dat]: sorry! done.

File: debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/J9BCUtil.java [hunk 2]
```diff
@@ -311,6 +311,10 @@ private static void dumpModifiers(PrintStream out, long modifiers, int modifierS
 					out.append("(constant) ");
 					modifiers &= ~J9FieldFlags.J9FieldFlagConstant;
 				}
+				if ((modifiers & J9FieldFlags.J9FieldFlagIsNullRestricted) != 0) {
+					out.append("(NullRestricted) ");
+					modifiers &= ~J9FieldFlags.J9FieldFlagIsNullRestricted;
+				}
```
Comment (keithc-ca) [inline debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/J9BCUtil.java:317]: I think this is only appropriate for instance fields; `J9StaticFieldIsNullRestricted` (which has a different value than `J9FieldFlagIsNullRestricted`) should be checked for static fields.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/J9BCUtil.java (hunk 2 above)
Comment (theresa-m) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/J9BCUtil.java:317]: That is for RAM classes, for ROM classes they both use `modifiers`.

Comment (keithc-ca) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/J9BCUtil.java:317]: We're talking about fields here, not classes, but to confirm, all null-restricted fields have `J9FieldFlagIsNullRestricted` included in `J9ROMFieldShape.modifiers` independent of whether the field is static or not?

File: debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/J9BCUtil.java [hunk 3]
```diff
@@ -261,7 +261,7 @@ private static void dumpModifiers(PrintStream out, long modifiers, int modifierS
 				if (ONLY_SPEC_MODIFIERS == modScope) {
 					modifiers &= J9CfrClassFile.CFR_FIELD_ACCESS_MASK;
 				} else {
-					modifiers &= J9CfrClassFile.CFR_FIELD_ACCESS_MASK | J9FieldFlags.J9FieldFlagConstant;
+					modifiers &= J9CfrClassFile.CFR_FIELD_ACCESS_MASK | J9FieldFlags.J9FieldFlagConstant | J9FieldFlags.J9FieldFlagIsNullRestricted;
```
Comment (keithc-ca) [inline debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/J9BCUtil.java:264]: Please update the comment for this method so it mentions `J9FieldFlagConstant` and `J9FieldFlagIsNullRestricted` in the set of valid modifiers for fields.

Comment (theresa-m) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/J9BCUtil.java:266]: updated.

File: debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/J9BCUtil.java (hunk 2 above)
Comment (theresa-m) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/J9BCUtil.java:319]: Yes. All fields written into the ROM class will have `J9FieldFlagIsNullRestricted ` in modifiers, the `J9StaticFieldIsNullRestricted ` field applies to resolved fields.

**Comment by keithc-ca** [comment]
Jenkins test sanity alinux64 jdk11,jdk21

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**